internal/manifests/collector/container.go

```diff
@@ -36,14 +38,26 @@ import (
 const maxPortLen = 15
 
 // Container builds a container for the given collector.
-func Container(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemetryCollector, addConfig bool) corev1.Container {
+func Container(cfg config.Config, logger logr.Logger, otelcol v1alpha2.OpenTelemetryCollector, addConfig bool) corev1.Container {
 	image := otelcol.Spec.Image
 	if len(image) == 0 {
 		image = cfg.CollectorImage()
 	}
 
+	configJson, err := json.Marshal(&otelcol.Spec.Config)
+	if err != nil {
+		logger.Error(err, "could not marshal config to json")
+		return corev1.Container{}
+	}
+
+	configYaml, err := yaml.JSONToYAML(configJson)
+	if err != nil {
+		logger.Error(err, "could not convert json to yaml")
+		return corev1.Container{}
+	}
+
 	// build container ports from service ports
-	ports, err := getConfigContainerPorts(logger, otelcol.Spec.Config)
+	ports, err := getConfigContainerPorts(logger, string(configYaml))
```

Comment:
should we just have this work off of the config type? Or is this something we want to do in a followup?

Reply:
Thats something I would like to do as a follow up. Originally I planned here also to submit a smaller PR. I would prefer to slowly work across the code base and do more intense changes in separate Prs.